### ISSUE TITLE
libcurl built on iOS11 Xcode9, fails to build because connectx is unavailable on iOS8

### ISSUE THREAD

**Issue by bitbytedog** [Cogosense/iOSCurlFramework#2]
xcode 11 checks all supported versions of ios for presence of functions. connectx is unavailable in iOS8.

This is a known issue in curl https://github.com/curl/curl/issues/1330.

**Comment by bitbytedog** [comment]
This is not a simple fix because the static configuration system cannot cope with backward compatibility.

The probable correct solution is to provide a weakly linked version of connectx that calls connect in the case that connectx is not available in iOS.

The actual solution was just to suppress the warning so curl would compile. Most probably support for iOS8 devices will not be required, we are already at iOS11.

**Comment by bitbytedog** [comment]
That didn't work, Jenkins build error:

~~~~
[armv7] /Users/jenkins/PRD/workspace/nse_iOSCurlFramework_master-Z6HZRBAR2KYHC32ZNRGG6L6QPNMD3BUONQLYHSY3HVN2O6RBUYWQ@2/build/curl-7.55.1/lib/connect.c:1069:12: error: 'connectx' is only available on iOS 9.0 or newer [-Werror,-Wunguarded-availability]
[armv7]       rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
[armv7]            ^~~~~~~~
[armv7] /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk/usr/include/sys/socket.h:713:5: note: 'connectx' has been explicitly marked partial here
[armv7] int connectx(int, const sa_endpoints_t *, sae_associd_t, unsigned int,
[armv7]     ^
[armv7] /Users/jenkins/PRD/workspace/nse_iOSCurlFramework_master-Z6HZRBAR2KYHC32ZNRGG6L6QPNMD3BUONQLYHSY3HVN2O6RBUYWQ@2/build/curl-7.55.1/lib/connect.c:1069:12: note: enclose 'connectx' in a __builtin_available check to silence this warning
[armv7]       rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
[armv7]            ^~~~~~~~
[armv7] 1 error generated.

~~~~

Followed suggestion in error message:

~~~~
 #if __has_builtin(connectx)
       rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
                     CONNECT_RESUME_ON_READ_WRITE | CONNECT_DATA_IDEMPOTENT,
                     NULL, 0, NULL, NULL);
 #else
       rc = connect(sockfd, &addr.sa_addr, addr.addrlen);
 #endif
~~~~
